### DISCUSSION TITLE
Add liboqs as a git submodule?

### DISCUSSION THREAD

**Discussion by baentsch** [open-quantum-safe/openssl#349] [Ideas]
github actions on oqs-openssl have been disabled to not trigger the normal OpenSSL github tests. Prime reason was that oqs-openssl doesn't build without `liboqs` (ie, the tests fail). So what about adding `liboqs` as a git submodule to `oqs-openssl` and enable "standalone build"? This might allow re-activation of OSSL github tests and ease external "liboqs-integrated openssl builds", e.g., in [msquic](https://github.com/microsoft/msquic)

**Comment by dstebila** [comment]
I am somewhat cautious about submodules because they are finicky to clone. But if there's a good reason, go ahead.

**Reply by baentsch** [reply]
Thanks for the experience report: "finicky" I don't like. A submodule would have simplified [this OQS-QUIC merge code](https://github.com/open-quantum-safe/oqs-demos/blob/main/quic/merge-oqs-openssl-quic.sh) -- but the gain is really small. So not worth while pursuing.

**Reply by vsoftco** [reply]
It's not just cloning, it's also about releases. When you tag a release, the submodule is not included in the .tgz/.zip. file. Users are asking Github for this feature for at least 3-4 years, but it's still not available.
